src/rust/lib/graph-editor/src/component/visualization/container.rs

```diff
@@ -47,49 +146,58 @@ impl Default for ContainerFrp {
 /// Container that wraps a `Visualization` for rendering and interaction in the GUI.
 ///
 /// The API to interact with the visualization is exposed through the `ContainerFrp`.
-#[derive(Clone,CloneRef,Debug,Shrinkwrap)]
+#[derive(Clone,CloneRef,Debug,Derivative,Shrinkwrap)]
+#[derivative(PartialEq)]
 #[allow(missing_docs)]
 pub struct Container {
     // The internals are split into two structs: `ContainerData` and `ContainerFrp`. The
     // `ContainerData` contains the actual data and logic for the `Container`. The `ContainerFrp`
     // contains the FRP api and network. This split is required to avoid creating cycles in the FRP
     // network: the FRP network holds `Rc`s to the `ContainerData` and thus must not live in the
     // same struct.
-
+    #[derivative(PartialEq(compare_with="Rc::ptr_eq"))]
     #[shrinkwrap(main_field)]
-        data : Rc<ContainerData>,
+    pub data : Rc<ContainerData>,
+    #[derivative(PartialEq="ignore")]
     pub frp  : ContainerFrp,
 }
 
 /// Internal data of a `Container`.
 #[derive(Debug,Clone)]
 #[allow(missing_docs)]
 pub struct ContainerData {
-    logger         : Logger,
-    display_object : display::object::Instance,
-    size           : Cell<Vector2<f32>>,
-    visualization  : RefCell<Option<Visualization>>,
+    logger                       : Logger,
+    size                         : Cell<Vector2<f32>>,
+    padding                      : Cell<f32>,
+    /// Topmost display object in the hierarchy. Used for global positioning.
+    display_object               : display::object::Instance,
+    /// Internal display object that will be sole child of `display_object` and can be attached and
+    /// detached from its parent for showing/hiding all child shapes.
+    display_object_internal      : display::object::Instance,
+    /// Parent of the visualisation. Allows adding/removing of visualisations without affecting
+    /// the order of other container shapes.
+    display_object_visualisation : display::object::Instance,
+
```

Comment:
The documentation for these 3 display objexcts is complex and tbh, I don't fully understand it. Is there a way to make the relations and responsibilities of the objects easier to understand?